File: fm-domain/src/main/java/io/zeitwert/fm/dms/model/ObjDocumentRepository.java
package io.zeitwert.fm.dms.model;

import io.dddrive.core.obj.model.ObjRepository;
import io.zeitwert.fm.dms.model.enums.CodeContentType;

public interface ObjDocumentRepository extends ObjRepository<ObjDocument> {

	byte[] getContent(ObjDocument document);

	CodeContentType getContentType(ObjDocument document);

	void storeContent(ObjDocument document, CodeContentType contentType, byte[] content);

}

File: fm-domain/src/main/java/io/zeitwert/fm/dms/model/ObjDocumentRepository.kt
package io.zeitwert.fm.dms.model

import dddrive.app.obj.model.ObjRepository
import io.zeitwert.fm.dms.model.enums.CodeContentType
import java.time.OffsetDateTime

interface ObjDocumentRepository : ObjRepository<ObjDocument> {

	fun getContent(document: ObjDocument): ByteArray?

	fun getContentType(document: ObjDocument): CodeContentType?

	fun storeContent(
		document: ObjDocument,
		contentType: CodeContentType,
		content: ByteArray,
		userId: Any,
		timestamp: OffsetDateTime,
	)

}
